File: TypeScript/src/repository/taskRepository.ts
import { Task, TaskModel } from "../models/taskModel";

export class TaskRepository {
    static async createNewTask(title: string, description: string, category: number, dueDate: string, user: string): Promise<boolean> {
        const createTaskResult = await TaskModel.createTask(title, description, category, dueDate, user);
        return createTaskResult;
    }

    static async fetchUserTasks(user: string): Promise<Task[] | null | undefined> {
        const userTasks = await TaskModel.readTasks(user);
        if (userTasks !== undefined) {
            return userTasks ? userTasks : null;
        } else {
            return undefined;
        }
    }

    static async fetchUserTaskById(taskId: number, userName: string) {
        const userTask = await TaskModel.fetchTaskByTaskId(taskId, userName);
        if (userTask !== undefined) {
            return userTask ? userTask : null;
        } else {
            return undefined;
        }
    }

    static async updateTask(taskId: number, taskTitle: string, taskDescription: string, taskCategory: number, taskDueDate: string, taskStatus: number, user: string): Promise<boolean> {
        const result = await TaskModel.updateTask(taskId, taskTitle, taskDescription, taskCategory, taskDueDate, taskStatus, user)
        if (result) {
            return result
        } else {
            return result
        }
    }

    static async deleteTask(taskId: number, userId: string): Promise<boolean> {
        const result = await TaskModel.deleteTask(taskId, userId);
        
        return result ? true : false
    }
}

File: C#/Repositories/TaskRepository.cs
using System;
using TaskManagerApp.Database;
using TaskManagerApp.Models;


namespace TaskManagerApp.Repositories
{
    public class TaskRepository
    {
        public static ICollection<Tasks> FetchUsersTasks(string username, AppDbContext context)
        {
            ICollection<Tasks> result = Models.TaskModel.FetchTasks(username, context);
            return result;
        }

        public static bool AddNewTask(string title, string description, string category, string dueDate, string status, string user, AppDbContext context)
        {
            bool result = Models.TaskModel.AddNewTask(title, description, category, dueDate, status, user, context);
            return result;
        }
    }
}